In [1]:
import pandas as pd
import pickle

In [2]:
full_data = pd.read_csv('full-data.csv')

In [3]:
full_data.describe()

Unnamed: 0     car_start       car_end      car_size  \
count  1.353067e+07  1.353067e+07  1.353067e+07  1.353067e+07   
mean   2.133779e+04  7.839622e+07  7.839675e+07  5.293201e+02   
std    1.740267e+04  5.725604e+07  5.725604e+07  3.808557e+02   
min    0.000000e+00  1.565000e+03  1.721000e+03  9.000000e+01   
25%    7.875000e+03  3.262257e+07  3.262288e+07  2.640000e+02   
50%    1.746400e+04  6.676723e+07  6.676805e+07  4.150000e+02   
75%    3.056200e+04  1.161726e+08  1.161737e+08  6.890000e+02   
max    1.244880e+05  2.492402e+08  2.492407e+08  3.475000e+04   

       car_intlogqvalue  car_foldchange    car_pvalue    car_qvalue  \
count      1.353067e+07    1.353067e+07  1.353067e+07  1.353067e+07   
mean       3.735343e+02    8.600618e+00  4.030745e+01  3.740348e+01   
std        6.916966e+02    4.540406e+00  6.946751e+01  6.916963e+01   
min        2.000000e+01    1.291750e+00  3.528590e+00  2.000010e+00   
25%        7.900000e+01    5.254350e+00  1.062482e+01  7.992097e+00   
50%        1.610000e+02    7.381760e+00  1.902249e+01  1.618277e+01   
75%        3.840000e+02    1.079550e+01  4.158542e+01  3.847557e+01   
max        3.940020e+05    6.767961e+01  3.940971e+04  3.940022e+04   

       car_summit_to_start  car_sample_id  car_region_id  car_risk_snp_anno  \
count         1.353067e+07   1.353067e+07   1.353067e+07       1.353067e+07   
mean          2.608023e+02   5.908581e+02   2.133879e+04       5.375677e-02   
std           2.324874e+02   4.334217e+02   1.740267e+04       2.971480e-01   
min           0.000000e+00   2.500000e+01   1.000000e+00       0.000000e+00   
25%           1.090000e+02   2.470000e+02   7.876000e+03       0.000000e+00   
50%           1.930000e+02   3.340000e+02   1.746500e+04       0.000000e+00   
75%           3.390000e+02   1.001000e+03   3.056300e+04       0.000000e+00   
max           3.307100e+04   1.471000e+03   1.244890e+05       4.200000e+01   

       car_eqtl_anno  car_crisps_anno   car_te_anno   car_se_anno  \
count   1.353067e+07     1.353067e+07  1.353067e+07  1.353067e+07   
mean    8.810028e-01     5.671376e+00  7.547725e+01  1.584883e+01   
std     1.728382e+00     3.000744e+01  8.208367e+01  3.374650e+01   
min     0.000000e+00     0.000000e+00  0.000000e+00  0.000000e+00   
25%     0.000000e+00     0.000000e+00  1.600000e+01  0.000000e+00   
50%     0.000000e+00     0.000000e+00  5.000000e+01  3.000000e+00   
75%     1.000000e+00     0.000000e+00  1.030000e+02  1.600000e+01   
max     1.800000e+02     9.060000e+02  4.960000e+02  4.630000e+02   

       car_methylation_anno  car_interaction_anno  car_commmon_snp_anno  \
count          1.353067e+07          1.353067e+07          1.353067e+07   
mean           1.479947e+02          5.284292e+02          7.206489e+00   
std            2.349735e+02          6.466926e+02          6.904346e+00   
min            0.000000e+00          0.000000e+00          0.000000e+00   
25%            0.000000e+00          3.600000e+01          3.000000e+00   
50%            0.000000e+00          2.150000e+02          5.000000e+00   
75%            2.520000e+02          9.660000e+02          1.000000e+01   
max            4.599000e+03          3.846200e+04          8.920000e+02   

       Region number  
count   1.353067e+07  
mean    4.267657e+04  
std     2.128934e+04  
min     2.500000e+01  
25%     2.875600e+04  
50%     4.054900e+04  
75%     5.556900e+04  
max     1.244890e+05

In [4]:
full_data.head()

Unnamed: 0 car_chr  car_start  car_end  car_size  car_intlogqvalue  \
0           0    chr1       9997    10802       805              1359   
1           1    chr1      20963    21340       377               108   
2           2    chr1      28785    29747       962              1636   
3           3    chr1     437449   437967       518               106   
4           4    chr1     540776   541301       525               263   

   car_foldchange  car_pvalue  car_qvalue  car_summit_to_start  ...  \
0        13.36316   138.88042   135.96225                  204  ...   
1         5.73316    13.03911    10.84643                  329  ...   
2        18.39825   166.64635   163.63165                  551  ...   
3         6.49730    12.88990    10.69930                  515  ...   
4         9.65766    28.77254    26.39556                  193  ...   

   car_se_anno  car_methylation_anno car_interaction_anno  \
0            0                     0                    1   
1            0                     0                    0   
2            0                   189                    3   
3            0                     0                    0   
4            2                     0                    6   

   car_commmon_snp_anno    Sample ID  Biosample type  \
0                     5  Sample_0025    Primary Cell   
1                     0  Sample_0025    Primary Cell   
2                     1  Sample_0025    Primary Cell   
3                     0  Sample_0025    Primary Cell   
4                     4  Sample_0025    Primary Cell   

                   Biosample name  Tissue type  Cancer type  Region number  
0  CD56Bright natural killer cell        Blood       Normal          48025  
1  CD56Bright natural killer cell        Blood       Normal          48025  
2  CD56Bright natural killer cell        Blood       Normal          48025  
3  CD56Bright natural killer cell        Blood       Normal          48025  
4  CD56Bright natural killer cell        Blood       Normal          48025  

[5 rows x 27 columns]

In [5]:
full_data[full_data['car_chr'] == 'chr2']

Unnamed: 0 car_chr  car_start    car_end  car_size  \
23315          23315    chr2      10739      11145       406   
23316          23316    chr2      11441      11898       457   
23317          23317    chr2      18282      18757       475   
23318          23318    chr2      46773      47240       467   
23319          23319    chr2     176111     176702       591   
...              ...     ...        ...        ...       ...   
13529744        1373    chr2  241525827  241526055       228   
13529745        1374    chr2  241916518  241917263       745   
13529746        1375    chr2  241917621  241920195      2574   
13529747        1376    chr2  242674001  242674229       228   
13529748        1377    chr2  242863861  242864128       267   

          car_intlogqvalue  car_foldchange  car_pvalue  car_qvalue  \
23315                  127         6.69815    14.97607    12.75303   
23316                  219         9.12304    24.25192    21.91752   
23317                  655        16.87831    68.17792    65.54793   
23318                  146         6.76221    16.88196    14.63372   
23319                  852        16.71831    87.94867    85.22743   
...                    ...             ...         ...         ...   
13529744                90         8.12534    12.92053     9.07606   
13529745                66         6.41960    10.40724     6.67950   
13529746                87         7.27317    12.54394     8.71652   
13529747                51         6.14061     8.82423     5.19908   
13529748                33         4.88382     6.78839     3.35025   

          car_summit_to_start  ...  car_se_anno  car_methylation_anno  \
23315                     210  ...            0                     0   
23316                     198  ...            0                     0   
23317                     260  ...            1                     0   
23318                     159  ...            7                   126   
23319                     449  ...            0                     0   
...                       ...  ...          ...                   ...   
13529744                  181  ...            5                     0   
13529745                  427  ...            6                     0   
13529746                  912  ...            6                     0   
13529747                   95  ...            2                     0   
13529748                  115  ...            1                     0   

         car_interaction_anno  car_commmon_snp_anno    Sample ID  \
23315                       0                    11  Sample_0025   
23316                       0                    22  Sample_0025   
23317                       0                     7  Sample_0025   
23318                     504                     8  Sample_0025   
23319                     226                     9  Sample_0025   
...                       ...                   ...          ...   
13529744                  506                     4  Sample_1471   
13529745                    3                     0  Sample_1471   
13529746                    8                     4  Sample_1471   
13529747                 1157                     0  Sample_1471   
13529748                   13                     0  Sample_1471   

          Biosample type                  Biosample name  Tissue type  \
23315       Primary Cell  CD56Bright natural killer cell        Blood   
23316       Primary Cell  CD56Bright natural killer cell        Blood   
23317       Primary Cell  CD56Bright natural killer cell        Blood   
23318       Primary Cell  CD56Bright natural killer cell        Blood   
23319       Primary Cell  CD56Bright natural killer cell        Blood   
...                  ...                             ...          ...   
13529744           Other                     Neutrophile        Blood   
13529745           Other                     Neutrophile        Blood   
13529746           Other                     Neut

In [6]:
genome = pickle.load(open('hg19.pickle', 'rb'))

In [7]:
print(genome.keys())

dict_keys(['chr1', 'chr2', 'chr3', 'chr4', 'chr5', 'chr6', 'chr7', 'chr8', 'chr9', 'chr10', 'chr11', 'chr12', 'chr13', 'chr14', 'chr15', 'chr16', 'chr17', 'chr18', 'chr19', 'chr20', 'chr21', 'chr22', 'chrX', 'chrY'])


In [8]:
[len(genome[k]) for k in genome.keys()]

[249250621,
 243199373,
 198022430,
 191154276,
 180915260,
 171115067,
 159138663,
 146364022,
 141213431,
 135534747,
 135006516,
 133851895,
 115169878,
 107349540,
 102531392,
 90354753,
 81195210,
 78077248,
 59128983,
 63025520,
 48129895,
 51304566,
 155270560,
 59373566]

In [9]:
genome_size = sum([len(genome[k]) for k in genome.keys()])
print(genome_size)

3095677412


In [10]:
genome_size / 10000

309567.7412

seems like a bucket size of 10k gives a reasonable number of 309568 buckets

In [11]:
experimental_full = full_data.copy(deep=True)

In [12]:
chrom_sizes = [len(genome[k]) for k in genome.keys()]

In [13]:
starting_indices = [sum(chrom_sizes[:idx]) for idx, item in enumerate(chrom_sizes)]

In [14]:
print(starting_indices)

[0, 249250621, 492449994, 690472424, 881626700, 1062541960, 1233657027, 1392795690, 1539159712, 1680373143, 1815907890, 1950914406, 2084766301, 2199936179, 2307285719, 2409817111, 2500171864, 2581367074, 2659444322, 2718573305, 2781598825, 2829728720, 2881033286, 3036303846]


In [15]:
starting_indices[-1] + chrom_sizes[-1]

3095677412

In [16]:
indices_by_chrom = {k:si for k, si in zip(genome.keys(), starting_indices)}

In [17]:
print(indices_by_chrom)

{'chr1': 0, 'chr2': 249250621, 'chr3': 492449994, 'chr4': 690472424, 'chr5': 881626700, 'chr6': 1062541960, 'chr7': 1233657027, 'chr8': 1392795690, 'chr9': 1539159712, 'chr10': 1680373143, 'chr11': 1815907890, 'chr12': 1950914406, 'chr13': 2084766301, 'chr14': 2199936179, 'chr15': 2307285719, 'chr16': 2409817111, 'chr17': 2500171864, 'chr18': 2581367074, 'chr19': 2659444322, 'chr20': 2718573305, 'chr21': 2781598825, 'chr22': 2829728720, 'chrX': 2881033286, 'chrY': 3036303846}


In [20]:
from pandarallel import pandarallel

pandarallel.initialize(progress_bar=True, nb_workers=14)


def transform_row_indices(row):
    new_row = row.copy()
    chromosome = row['car_chr']
    start = row['car_start']
    end = row['car_end']
    
    new_row['car_start'] = start + indices_by_chrom[chromosome]
    new_row['car_end'] = end + indices_by_chrom[chromosome]
    return new_row

experimental_full = experimental_full.parallel_apply(transform_row_indices, axis=1)

INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.


In [23]:
experimental_full[experimental_full['car_chr'] == 'chr21']

Unnamed: 0 car_chr   car_start     car_end  car_size  \
28311          28311   chr21  2791553577  2791554111       534   
28312          28312   chr21  2791762365  2791763127       762   
28313          28313   chr21  2792195419  2792196221       802   
28314          28314   chr21  2792726327  2792726844       517   
28315          28315   chr21  2792744351  2792744915       564   
...              ...     ...         ...         ...       ...   
13529840        1469   chr21  2829203245  2829203459       214   
13529841        1470   chr21  2829311547  2829313285      1738   
13529842        1471   chr21  2829342611  2829342863       252   
13529843        1472   chr21  2829477383  2829477590       207   
13529844        1473   chr21  2829654181  2829654438       257   

          car_intlogqvalue  car_foldchange  car_pvalue  car_qvalue  \
28311                   60         4.87297     8.19386     6.09575   
28312                 1018        16.39443   104.66055   101.87089   
28313                   83         5.23280    10.53824     8.39181   
28314                   44         4.22324     6.49794     4.44450   
28315                   21         3.09952     4.13673     2.17799   
...                    ...             ...         ...         ...   
13529840               172        10.89684    21.29443    17.21299   
13529841                75         7.32590    11.27761     7.50508   
13529842               120         9.15432    16.04107    12.09240   
13529843               146        10.68906    18.68340    14.66471   
13529844                91         8.17399    13.00703     9.15876   

          car_summit_to_start  ...  car_se_anno  car_methylation_anno  \
28311                     468  ...            0                     0   
28312                     263  ...            0                     0   
28313                     310  ...            0                     0   
28314                     180  ...            3                     0   
28315                     368  ...            4                     0   
...                       ...  ...          ...                   ...   
13529840                  133  ...            2                     0   
13529841                 1577  ...            0                     0   
13529842                  141  ...            6                   315   
13529843                   82  ...            3                   126   
13529844                   72  ...            5                   126   

         car_interaction_anno  car_commmon_snp_anno    Sample ID  \
28311                       0                    11  Sample_0025   
28312                       1                     0  Sample_0025   
28313                       1                     6  Sample_0025   
28314                      34                     1  Sample_0025   
28315                      22                     0  Sample_0025   
...                       ...                   ...          ...   
13529840                  821                     0  Sample_1471   
13529841                   32                     7  Sample_1471   
13529842                 1180                    10  Sample_1471   
13529843                 1193                     2  Sample_1471   
13529844                 1139                    11  Sample_1471   

          Biosample type                  Biosample name  Tissue type  \
28311       Primary Cell  CD56Bright natural killer cell        Blood   
28312       Primary Cell  CD56Bright natural killer cell        Blood   
28313       Primary Cell  CD56Bright natural killer cell        Blood   
28314       Primary Cell  CD56Bright natural killer cell        Blood   
28315       Primary Cell  CD56Bright natural killer cell        Blood   
...                  ...                             ...          ...   
13529840           Other                     Neutrophile        Blood   
13529841           Other                     Neutrophile        Blood   
13529842           Other 

In [52]:
from pandarallel import pandarallel
import math
import numpy as np

pandarallel.initialize(progress_bar=True, nb_workers=14)

binsize = 10000

num_bins = math.ceil(3095677412 / binsize)
print(num_bins)

sample_25_seqs = experimental_full[experimental_full['Sample ID'] == 'Sample_0025']
sample_25_acc = pd.DataFrame({'Sample ID': 'Sample_0025', 'bin num': 0, 'accessibility %': 0}, index=np.arange(num_bins), columns=['Sample ID', 'bin num', 'accessibility %'])
display(sample_25_acc.head())


def calculate_accessibility(row):
    new_row = row.copy()
    mybin = row.name
    b_start = mybin * binsize
    b_end = (mybin+1) * binsize
    
    asdf = sample_25_seqs[sample_25_seqs['car_end'] >= b_start]
    asdf = asdf[asdf['car_start'] <= b_end]
    
    def helper_f(hrow):
        newh = hrow.copy()
        if hrow['car_start'] < b_start and hrow['car_end'] > b_end:
            newh['contrib'] = binsize
        elif hrow['car_start'] < b_start:
            newh['contrib'] = hrow['car_end'] - b_start
        elif hrow['car_end'] > b_end:
            newh['contrib'] = b_end - hrow['car_start']
        else:
            newh['contrib'] = hrow['car_end'] - hrow['car_start']
        return newh
    
    asdf = asdf.apply(helper_f, axis=1)
    
    new_row['bin num'] = row.name
    if 'contrib' in asdf.columns:
        new_row['accessibility %'] = asdf['contrib'].sum() / binsize
    else:
        new_row['accessibility %'] = 0
    return new_row

abc = sample_25_acc.parallel_apply(calculate_accessibility, axis=1)

INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


Sample ID  bin num  accessibility %
0  Sample_0025        0                0
1  Sample_0025        0                0
2  Sample_0025        0                0
3  Sample_0025        0                0
4  Sample_0025        0                0

In [59]:
abc.head()

Sample ID  bin num  accessibility %
0  Sample_0025        0           0.0003
1  Sample_0025        1           0.0802
2  Sample_0025        2           0.1339
3  Sample_0025        3           0.0000
4  Sample_0025        4           0.0000

In [55]:
abc.describe()

bin num  accessibility %
count  309568.000000    309568.000000
mean   154783.500000         0.010719
std     89364.728404         0.031997
min         0.000000         0.000000
25%     77391.750000         0.000000
50%    154783.500000         0.000000
75%    232175.250000         0.000000
max    309567.000000         0.746100

In [56]:
abc[abc['accessibility %'] > 0].describe()

bin num  accessibility %
count   42502.000000     42502.000000
mean   154713.007106         0.078073
std     89951.363558         0.046888
min         0.000000         0.000100
25%     72848.250000         0.042800
50%    162603.500000         0.068900
75%    239057.500000         0.102800
max    309566.000000         0.746100

In [77]:
from pandarallel import pandarallel
import math
import numpy as np

def process_one_sample(s_id):

    pandarallel.initialize(progress_bar=True, nb_workers=14)

    binsize = 10000

    num_bins = math.ceil(3095677412 / binsize)
    print(num_bins)

    sample_seqs = experimental_full[experimental_full['Sample ID'] == s_id]
    sample_acc = pd.DataFrame({'Sample ID': s_id, 'bin num': 0, 'accessibility %': 0}, index=np.arange(num_bins), columns=['Sample ID', 'bin num', 'accessibility %'])


    def calculate_accessibility(row):
        new_row = row.copy()
        mybin = row.name
        b_start = mybin * binsize
        b_end = (mybin+1) * binsize

        asdf = sample_seqs[sample_seqs['car_end'] >= b_start]
        asdf = asdf[asdf['car_start'] <= b_end]

        def helper_f(hrow):
            newh = hrow.copy()
            if hrow['car_start'] < b_start and hrow['car_end'] > b_end:
                newh['contrib'] = binsize
            elif hrow['car_start'] < b_start:
                newh['contrib'] = hrow['car_end'] - b_start
            elif hrow['car_end'] > b_end:
                newh['contrib'] = b_end - hrow['car_start']
            else:
                newh['contrib'] = hrow['car_end'] - hrow['car_start']
            return newh

        asdf = asdf.apply(helper_f, axis=1)

        new_row['bin num'] = row.name
        if 'contrib' in asdf.columns:
            new_row['accessibility %'] = asdf['contrib'].sum() / binsize
        else:
            new_row['accessibility %'] = 0
        return new_row

    abc = sample_acc.parallel_apply(calculate_accessibility, axis=1)
    return abc['accessibility %']

In [78]:
len(experimental_full['Sample ID'].unique())

452

In [83]:
binsize = 10000
num_bins = math.ceil(3095677412 / binsize)

collection_of_new_data = pd.DataFrame(columns=[str(n) for n in range(num_bins)])
print(collection_of_new_data)


stuff = list()

for sampleid in experimental_full['Sample ID'].unique():
    percents = process_one_sample(sampleid)
    percents.name = sampleid
    stuff.append(percents)

Empty DataFrame
Columns: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87, 88, 89, 90, 91, 92, 93, 94, 95, 96, 97, 98, 99, ...]
Index: []

[0 rows x 309568 columns]
INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


INFO: Pandarallel will run on 14 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.
309568


In [72]:
finaldf = pd.DataFrame(stuff)
display(finaldf)

0       1       2       3       4       5       6       7       8       \
0  0.0003  0.0802  0.1339     0.0     0.0     0.0     0.0     0.0     0.0   

   9       ...  309558  309559  309560  309561  309562  309563  309564  \
0     0.0  ...     0.0     0.0     0.0     0.0     0.0   0.062  0.0243   

   309565  309566  309567  
0  0.0593  0.0535     0.0  

[1 rows x 309568 columns]

In [84]:
finaldf = pd.concat(stuff, axis=1).transpose()
display(finaldf)

0       1       2       3       4       5       6       7       \
Sample_0025  0.0003  0.0802  0.1339     0.0     0.0     0.0     0.0     0.0   
Sample_0026  0.0003  0.1179  0.0973     0.0     0.0     0.0     0.0     0.0   
Sample_0027  0.0000  0.0653  0.0591     0.0     0.0     0.0     0.0     0.0   
Sample_0028  0.0000  0.0768  0.0515     0.0     0.0     0.0     0.0     0.0   
Sample_0029  0.0003  0.1085  0.0850     0.0     0.0     0.0     0.0     0.0   
...             ...     ...     ...     ...     ...     ...     ...     ...   
Sample_1467  0.0000  0.0450  0.0659     0.0     0.0     0.0     0.0     0.0   
Sample_1468  0.0000  0.0450  0.0000     0.0     0.0     0.0     0.0     0.0   
Sample_1469  0.0000  0.0429  0.0697     0.0     0.0     0.0     0.0     0.0   
Sample_1470  0.0000  0.0440  0.0639     0.0     0.0     0.0     0.0     0.0   
Sample_1471  0.0000  0.0385  0.0000     0.0     0.0     0.0     0.0     0.0   

             8       9       ...  309558  309559  309560  309561  309562  \
Sample_0025     0.0     0.0  ...     0.0     0.0     0.0     0.0     0.0   
Sample_0026     0.0     0.0  ...     0.0     0.0     0.0     0.0     0.0   
Sample_0027     0.0     0.0  ...     0.0     0.0     0.0     0.0     0.0   
Sample_0028     0.0     0.0  ...     0.0     0.0     0.0     0.0     0.0   
Sample_0029     0.0     0.0  ...     0.0     0.0     0.0     0.0     0.0   
...             ...     ...  ...     ...     ...     ...     ...     ...   
Sample_1467     0.0     0.0  ...     0.0     0.0     0.0     0.0     0.0   
Sample_1468     0.0     0.0  ...     0.0     0.0     0.0     0.0     0.0   
Sample_1469     0.0     0.0  ...     0.0     0.0     0.0     0.0     0.0   
Sample_1470     0.0     0.0  ...     0.0     0.0     0.0     0.0     0.0   
Sample_1471     0.0     0.0  ...     0.0     0.0     0.0     0.0     0.0   

             309563  309564  309565  309566  309567  
Sample_0025  0.0620  0.0243  0.0593  0.0535     0.0  
Sample_0026  0.0609  0.0257  0.0618  0.0386     0.0  
Sample_0027  0.0421  0.0000  0.0000  0.0548     0.0  
Sample_0028  0.0601  0.0104  0.0131  0.0539     0.0  
Sample_0029  0.0419  0.0000  0.0000  0.0565     0.0  
...             ...     ...     ...     ...     ...  
Sample_1467  0.0000  0.0000  0.0000  0.0540     0.0  
Sample_1468  0.0000  0.0000  0.0000  0.0536     0.0  
Sample_1469  0.0000  0.0000  0.0000  0.0522     0.0  
Sample_1470  0.0000  0.0000  0.0000  0.0539     0.0  
Sample_1471  0.0000  0.0000  0.0000  0.0351     0.0  

[452 rows x 309568 columns]

In [98]:
persample = full_data[['Sample ID', 'Biosample type', 'Biosample name', 'Tissue type', 'Cancer type', 'Region number']]
persample = persample.drop_duplicates()
persample = persample.set_index('Sample ID')
persample.index.name = ''
display(persample)

Biosample type                  Biosample name Tissue type  \
                                                                         
Sample_0025   Primary Cell  CD56Bright natural killer cell       Blood   
Sample_0026   Primary Cell  CD56Bright natural killer cell       Blood   
Sample_0027   Primary Cell  CD56Bright natural killer cell       Blood   
Sample_0028   Primary Cell  CD56Bright natural killer cell       Blood   
Sample_0029   Primary Cell     CD56Dim natural killer cell       Blood   
...                    ...                             ...         ...   
Sample_1467          Other                      Macrophage       Blood   
Sample_1468          Other                        Monocyte       Blood   
Sample_1469          Other                        Monocyte       Blood   
Sample_1470          Other                     Neutrophile       Blood   
Sample_1471          Other                     Neutrophile       Blood   

            Cancer type  Region number  
                                        
Sample_0025      Normal          48025  
Sample_0026      Normal          50447  
Sample_0027      Normal          47349  
Sample_0028      Normal          46698  
Sample_0029      Normal          40745  
...                 ...            ...  
Sample_1467       Other          10568  
Sample_1468       Other           6102  
Sample_1469       Other           8042  
Sample_1470       Other           7806  
Sample_1471       Other           2297  

[452 rows x 5 columns]

In [103]:
superfinal = pd.concat([persample, finaldf], axis=1)

In [104]:
display(superfinal)

Biosample type                  Biosample name Tissue type  \
Sample_0025   Primary Cell  CD56Bright natural killer cell       Blood   
Sample_0026   Primary Cell  CD56Bright natural killer cell       Blood   
Sample_0027   Primary Cell  CD56Bright natural killer cell       Blood   
Sample_0028   Primary Cell  CD56Bright natural killer cell       Blood   
Sample_0029   Primary Cell     CD56Dim natural killer cell       Blood   
...                    ...                             ...         ...   
Sample_1467          Other                      Macrophage       Blood   
Sample_1468          Other                        Monocyte       Blood   
Sample_1469          Other                        Monocyte       Blood   
Sample_1470          Other                     Neutrophile       Blood   
Sample_1471          Other                     Neutrophile       Blood   

            Cancer type  Region number       0       1       2    3    4  ...  \
Sample_0025      Normal          48025  0.0003  0.0802  0.1339  0.0  0.0  ...   
Sample_0026      Normal          50447  0.0003  0.1179  0.0973  0.0  0.0  ...   
Sample_0027      Normal          47349  0.0000  0.0653  0.0591  0.0  0.0  ...   
Sample_0028      Normal          46698  0.0000  0.0768  0.0515  0.0  0.0  ...   
Sample_0029      Normal          40745  0.0003  0.1085  0.0850  0.0  0.0  ...   
...                 ...            ...     ...     ...     ...  ...  ...  ...   
Sample_1467       Other          10568  0.0000  0.0450  0.0659  0.0  0.0  ...   
Sample_1468       Other           6102  0.0000  0.0450  0.0000  0.0  0.0  ...   
Sample_1469       Other           8042  0.0000  0.0429  0.0697  0.0  0.0  ...   
Sample_1470       Other           7806  0.0000  0.0440  0.0639  0.0  0.0  ...   
Sample_1471       Other           2297  0.0000  0.0385  0.0000  0.0  0.0  ...   

             309558  309559  309560  309561  309562  309563  309564  309565  \
Sample_0025     0.0     0.0     0.0     0.0     0.0  0.0620  0.0243  0.0593   
Sample_0026     0.0     0.0     0.0     0.0     0.0  0.0609  0.0257  0.0618   
Sample_0027     0.0     0.0     0.0     0.0     0.0  0.0421  0.0000  0.0000   
Sample_0028     0.0     0.0     0.0     0.0     0.0  0.0601  0.0104  0.0131   
Sample_0029     0.0     0.0     0.0     0.0     0.0  0.0419  0.0000  0.0000   
...             ...     ...     ...     ...     ...     ...     ...     ...   
Sample_1467     0.0     0.0     0.0     0.0     0.0  0.0000  0.0000  0.0000   
Sample_1468     0.0     0.0     0.0     0.0     0.0  0.0000  0.0000  0.0000   
Sample_1469     0.0     0.0     0.0     0.0     0.0  0.0000  0.0000  0.0000   
Sample_1470     0.0     0.0     0.0     0.0     0.0  0.0000  0.0000  0.0000   
Sample_1471     0.0     0.0     0.0     0.0     0.0  0.0000  0.0000  0.0000   

             309566  309567  
Sample_0025  0.0535     0.0  
Sample_0026  0.0386     0.0  
Sample_0027  0.0548     0.0  
Sample_0028  0.0539     0.0  
Sample_0029  0.0565     0.0  
...             ...     ...  
Sample_1467  0.0540     0.0  
Sample_1468  0.0536     0.0  
Sample_1469  0.0522     0.0  
Sample_1470  0.0539     0.0  
Sample_1471  0.0351     0.0  

[452 rows x 309573 columns]

In [105]:
superfinal.to_csv('accessibilty-percents-10000-bin.csv')

# Data exploration